x-pack/platform/plugins/shared/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -38,15 +38,14 @@ import { getDegradedFieldValues } from './get_degraded_field_values';
 import { getDegradedFields } from './get_degraded_fields';
 import { getNonAggregatableDataStreams } from './get_non_aggregatable_data_streams';
 import { updateFieldLimit } from './update_field_limit';
+import { getDataStreamsCreationDate } from './get_data_streams_creation_date';
 
 const statsRoute = createDatasetQualityServerRoute({
   endpoint: 'GET /internal/dataset_quality/data_streams/stats',
   params: t.type({
     query: t.intersection([
-      t.type({ types: typesRt }),
-      t.partial({
-        datasetQuery: t.string,
-      }),
+      t.union([t.type({ types: typesRt }), t.type({ datasetQuery: t.string })]),
+      t.partial({ includeCreationDate: t.boolean }),
```

Comment:
I figured we either use `types` or `datasetQuery` and updated the type accordingly